tests/e2e/test_pv_scale_and_respin_ceph_pods.py

```diff
@@ -0,0 +1,184 @@
+        for pod_obj in temp_pod_objs:
+            process = threading.Thread(target=pod_obj.run_io, args=('fs', '512M', ))
+            process.start()
+            threads.append(process)
+        for pod_obj in rbd_rwx_pods:
+            process = threading.Thread(target=pod_obj.run_io, args=('block', '512M', ))
+            process.start()
+            threads.append(process)
+        for process in threads:
+            process.join()
+
+    def respin_ceph_pod(self, resource_to_delete):
+        """
+        Function to respin ceph pods one by one,
+        delete_resource functions checks for the deleted pod back up and running
+
+        Args:
+            resource_to_delete (str): Ceph resource type to be deleted, eg: mgr/mon/osd/mds
+        """
+        disruption = disruption_helpers.Disruptions()
+        disruption.set_resource(resource=resource_to_delete)
+        no_of_resource = disruption.resource_count
+        for i in range(0, no_of_resource):
+            disruption.delete_resource(resource_id=i)
+
+    def cleanup(self):
+        """
+        Function to cleanup the SC, PVC and POD objects parallel.
+        """
+        helpers.delete_objs_parallel(pod.get_all_pods(namespace=self.namespace))
+        helpers.delete_objs_parallel(self.all_pvc_obj)
+        self.rbd_sc_obj.delete()
+        self.cephfs_sc_obj.delete()
+
+
+@scale
+@pytest.mark.parametrize(
+    argnames="resource_to_delete",
+    argvalues=[
+        pytest.param(
+            *['mgr'], marks=pytest.mark.polarion_id("OCS-766")
+        ),
+        pytest.param(
+            *['mon'], marks=pytest.mark.polarion_id("OCS-764")
+        ),
+        pytest.param(
+            *['osd'], marks=pytest.mark.polarion_id("OCS-765")
+        ),
+        pytest.param(
+            *['mds'], marks=pytest.mark.polarion_id("OCS-613")
+        )
+    ]
+)
+class TestPVSTOcsCreatePVCsAndRespinCephPods(BasePvcCreateRespinCephPods):
+    """
+    Class for PV scale Create Cluster with 1000 PVC, then Respin ceph pods
+    Check for Memory leak, network and stats.
+    """
+    @pytest.fixture()
+    def setup_fixture(self, request):
+        def finalizer():
+            self.cleanup()
+
+        request.addfinalizer(finalizer)
+
+    @pytest.fixture()
+    def namespace(self, project_factory):
+        """
+        Create a project for the test
+        """
+        proj_obj = project_factory()
+        self.namespace = proj_obj.namespace
+
+    @pytest.fixture()
+    def storageclass(self, storageclass_factory):
+        """
+        Create Storage class for rbd and cephfs
+        """
+        self.rbd_sc_obj = storageclass_factory(interface=constants.CEPHBLOCKPOOL)
+        self.cephfs_sc_obj = storageclass_factory(interface=constants.CEPHFILESYSTEM)
+
+    def test_pv_scale_out_create_pvcs_and_respin_ceph_pods(
+        self, memory_leak_function, namespace, storageclass, setup_fixture, resource_to_delete
+    ):
+        self.pvc_count_each_itr = 10
```

Comment:
no need to use seld for these variables as they are not used outside this method

Reply:
ack